components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -697,10 +712,20 @@ abstract class AbstractFetchDownloadService : Service() {
 
             outStream.write(data, 0, bytesRead)
         }
+        if (isInStreamClosed) {
+            // In cases where [download.response] is available and users with slow
+            // networks start a download but quickly press pause and then resume
+            // [isResumingDownload] will be false as there will be not enough time
+            // for bytes to be copied, but the stream in [download.response] will be closed,
+            // we have to fallback to [httpClient]
```

Comment:
Currently this logic would also be applied for downloads already using the HTTP client. I think in this case we don't need to fall back and we don't need to call `performDownload` again as we're already using the same client anyway. I think we should check for that here...only fall back to the fetch client if we're not already using it.

Reply:
I also think it's worth filing this for GV to potentially improve the API.

Reply:
> Currently this logic would also be applied for downloads already using the HTTP client. I think in this case we don't need to fall back and we don't need to call performDownload again as we're already using the same client anyway. I think we should check for that here...only fall back to the fetch client if we're not already using it.

👍🏽 

> I also think it's worth filing this for GV to potentially improve the API.

👍🏽